GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -820,12 +823,39 @@ extension SecureConversations.TranscriptModel {
                 self.historySection.set(items)
                 self.action?(.refreshSection(self.historySection.index))
                 self.action?(.scrollToBottom(animated: false))
-               completion(messagesWithUnreadCount.messages)
+                completion(messagesWithUnreadCount.messages)
+                if messagesWithUnreadCount.unreadCount > 0 {
+                    self.markMessagesAsRead()
+                }
             case .failure:
                 completion([])
             }
         }
     }
+
+    func markMessagesAsRead() {
+        environment.gcd.mainQueue.asyncAfterDeadline(.now() + .seconds(Self.markUnreadMessagesDelaySeconds)) { [weak self] in
+            _ = self?.environment.secureMarkMessagesAsRead { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case .success:
+                    self.historySection.removeAll(where: {
+                        if case .unreadMessageDivider = $0.kind {
+                            return true
+                        }
+
+                        return false
+                    })
+                    self.action?(.refreshSection(self.historySection.index, animated: true))
+                case .failure:
+                    self.showAlert(
+                        with: self.environment.alertConfiguration.unexpectedError,
+                        dismissed: nil
+                    )
```

Comment:
would it be expected if the app will show an error if `messageRead` request failed? I would like to propose to fail silently since this is internal operation.

Reply:
Removed alert 967219a3fc777808296242efada693c9b5a209b4.